lib/ansible/modules/cloud/smartos/imgadm.py

```diff
@@ -0,0 +1,302 @@
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible. If not, see <http://www.gnu.org/licenses/>.
+#
+
+ANSIBLE_METADATA = {'status': ['preview'],
+                    'supported_by': 'community',
+                    'version': '1.0'}
+
+DOCUMENTATION = '''
+---
+module: imgadm
+short_description: Manage SmartOS images
+description:
+    - Manage SmartOS virtual machine images through imgadm(1M)
+version_added: "2.3"
+author: Jasper Lievisse Adriaanse (@jasperla)
+options:
+    force:
+        required: false
+        choices: [ yes, no ]
+        description:
+        - Force a given operation (where supported by imgadm(1M)).
+    pool:
+        required: false
+        default: zones
+        description:
+        - zpool to import to or delete images from.
+    source:
+        required: false
+        description:
+        - URI for the image source.
+    state:
+        required: true
+        choices: [ present, absent, deleted, imported, updated, vacumed ]
+        description:
+        - State the object operated on should be in. C(imported) is an alias for
+          for C(present) and C(deleted) for C(absent). When set to C(vacumed)
+          and C(uuid) to C(*), it will remove all unused images.
+    type:
+        required: false
+        choices: [ imgapi, docker, dsapi ]
+        default: imgapi
+        description:
+        - Type for image sources.
+    uuid:
+        required: false
+        description:
+        - Image UUID. Can either be a full UUID or C(*) for all images.
+'''
+
+EXAMPLES = '''
+- name: Import an image
+  imgadm:
+    uuid: '70e3ae72-96b6-11e6-9056-9737fd4d0764'
+    state: imported
+
+- name: Delete an image
+  imgadm:
+    uuid: '70e3ae72-96b6-11e6-9056-9737fd4d0764'
+    state: deleted
+
+- name: Update all images
+  imgadm:
+    uuid: '*'
+    state: updated
+
+- name: Update a single image
+  imgadm:
+    uuid: '70e3ae72-96b6-11e6-9056-9737fd4d0764'
+    state: updated
+
+- name: Add a source
+  imgadm:
+    source: 'https://datasets.project-fifo.net'
+    state: present
+
+- name: Add a Docker source
+  imgadm:
+    source: 'https://docker.io'
+    type: docker
+    state: present
+
+- name: Remove a source
+  imgadm:
+    source: 'https://docker.io'
+    state: absent
+'''
+
+import re
+
+# Shortcut for the imgadm(1M) command. While imgadm(1M) supports a
+# -E option to return any errors in JSON, the generated JSON does not play well
+# with the JSON parsers of Python. The returned message contains '\n' as part of
+# the stacktrace, which breaks the parsers.
+IMGADM = 'imgadm'
+
+# Helper method to massage stderr
+def errmsg(stderr):
+    match = re.match('^imgadm .*?: error \(\w+\): (.*): .*', stderr)
+    if match:
+        return match.groups()[0]
+    else:
+        return 'Unexpected failure'
+
+def update_images(module):
+    uuid = module.params['uuid']
+    cmd = IMGADM + ' update'
+
+    if uuid != '*':
+        cmd += ' {}'.format(uuid)
```

Comment:
Note: This syntax is python-2.7 or greater.  Using .format() with positions is 2.6+:
``` python
cmd += ' {0}'.format(uuid)
```

It's more pythonic to create a single string rather than two.  for instance:
``` python
cmd = '{0} {1}'.format(cmd, uuid)
# or
cmd = ' '.join((cmd, uuid))
```

If you need to be compatible with python less than 2.6, you need to avoid .format() altogether.  So the .join method or percent formatting:
``` python
cmd = '%s %s' % (cmd, uuid)
```